flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceWithWatermarkExtractor.java

```diff
@@ -92,7 +93,10 @@ public class TestIcebergSourceWithWatermarkExtractor implements Serializable {
               .setNumberTaskManagers(1)
               .setNumberSlotsPerTaskManager(PARALLELISM)
               .setRpcServiceSharing(RpcServiceSharing.DEDICATED)
-              .setConfiguration(reporter.addToConfiguration(new Configuration()))
+              .setConfiguration(
+                  reporter.addToConfiguration(
+                      // disable classloader check as Avro may cache class in the serializers.
+                      new Configuration().set(CoreOptions.CHECK_LEAKED_CLASSLOADER, false)))
```

Comment:
Would it worth/possible to have a common place for this, so we will not forget it in future tests?

Reply:
this was merged by PR #9216 . I included it earlier before the separate PR was merged